### ISSUE TITLE
configuration file to drop loginctl enable-linger snippnett

### ISSUE THREAD

**Issue by adrelanos** [systemd/systemd#3513]
`sudo loginctl enable-linger user` creates file `/var/lib/systemd/linger/user` but this as only option to enable lingering is non-ideal.

It would be great if there was a `.d` configuration folder where one could drop a configuration snippet to enable lingering.

**Comment by poettering** [comment]
hmm, wut? I don't follow? where should that directory be located? where's the advantage over just touching one file in /var/lib/systemd/linger? .d/ drop-ins are  a lot more complex than that...

**Comment by adrelanos** [comment]
Lennart Poettering:

> where should that directory be located?

Similar style like the other drop-ins.

/lib/systemd/config.d

/etc/systemd/config.d

> where's the advantage over just touching one file in
> /var/lib/systemd/linger?

Then packages by sysadmins and distributions can ship packages with
config files enabling this. `touch` is not appropriate. That does not
work when more than one package wants to enable this. When the package
that enabled lingering gets uninstalled, it would be unappropriated to
remove the lingering status file. Another package or the sysadmin might
want to keep it enabled.

**Comment by poettering** [comment]
hmm? lingering is a per-user thing (i.e. real-life user), why would a package manager bother with that? I mean, if I create a user "lennart" on my system I don't don#t do that by packaging some file into an RPM either...

Really not grokking what ou are asking for...

**Comment by adrelanos** [comment]
TLDR:

How to enable systemd --user services by default as a distribution using
a .d drop-in?

---

Long:

Perhaps this is indeed an xy-problem.

My actual use case is to replace an /etc/xdg/autostart/app.desktop with
a systemd --user service. /etc/xdg/autostart is not great, because it
does not support systemd `After=`.

No google search result revealed how one could enable systemd --user
services in Debian jessie.

The package maintainers did not reply:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=820111

The only way to do it was described as requiring lingering:
https://groups.google.com/d/msg/qubes-users/Verq_0ceE04/YgZxNjohQAAJ

As I understand lingering now, it just means that systemd services can
keep running even when the user has logged off. It is by no means a
requirement to make usual systemd --user services work. At least it
should not be.

**Comment by poettering** [comment]
I am sorry, but I don't see how this request makes sense. The "lingering" concept is a per-user thing. You enable it for user "foo" on a specific machine. It's _not_ a system-wide thing. If it was, then this would mean we'd end up starting systemd --user instances for _all_ local users at boot. But that's really not realistic. This must stay a per-user opt-in feature.

**Comment by poettering** [comment]
Closing, as I think this is really just a misunderstanding how lingering works.
